core/src/blocktree.rs

```diff
@@ -1062,10 +1064,6 @@ impl Blocktree {
 
                 blob_bytes.drain(..BLOB_HEADER_SIZE);
 
-                if size == 0 {
```

Comment:
this is the buggy clause.  if our coding_present looks like 0=false, 1=true, ... then the first coding buffer shard gets [0u8; 0] on line 1072 !